chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -245,10 +246,41 @@
       command "ovs-vsctl add-br #{name}; ip link set #{name} up"
       not_if "ovs-vsctl list-br |grep -q #{name}"
     end
-    execute "Neutron: add #{bound_if} to #{name}" do
-      command "ovs-vsctl del-port #{name} #{bound_if} ; ovs-vsctl add-port #{name} #{bound_if}"
-      not_if "ovs-dpctl show system@#{name} | grep -q #{bound_if}"
+
+    # If:
+    #   * this network is on the same physical interface as the admin network
+    #   * we're running in HA mode
+    #   * corosync is configured to use multicast networking
+    # We need to stop openais/corosync before moving the network interface
+    # into the openvswitch. Otherwise we'll break corosync's multicast
+    # communication and cause fencing.
+    # For details: https://bugzilla.suse.com/show_bug.cgi?id=915176
+    handle_corosync_restart = node[:crowbar_wall][:network][:nets]["admin"].last == bound_if &&
+        neutron_l3_ha && node[:corosync][:transport] == "udp"
+
+    not_if_clause = "ovs-vsctl iface-to-br #{bound_if} | grep -q #{name}"
+    if handle_corosync_restart
+      # Put the whole cluster into maintenance mode before shutting down
+      # corosync. As this operation affects all cluster nodes only do it on
+      # the founder.
+      if (CrowbarPacemakerHelper.is_cluster_founder?(node))
+        execute "enable maintenance mode" do
+          command "crm --wait configure property maintenance-mode=true"
+          not_if not_if_clause
+        end
+      end
+
+      crowbar_pacemaker_sync_mark "sync-neutron-l3_before_network_reconfig" do
+        not_if not_if_clause
+      end
+
+      service "stop openais for network reconfiguration" do
```

Comment:
nitpicking, I guess, but better use `node[:corosync][:platform][:service_name]` than `openais` in this string (same comment later on).